mono/profiler/decode.c

```diff
@@ -2400,6 +2400,21 @@ decode_buffer (ProfContext *ctx)
 					fprintf (outfile, "handle (%s) %u destroyed\n", get_handle_name (htype), handle);
 				if (frames != sframes)
 					free (frames);
+			} else if (subtype == TYPE_GC_FINALIZE_START) {
+				// TODO: Generate a finalizer report based on these events.
```

Comment:
I can think of 2 reports we'd care to have.

The format can be the same format as the alloc report, but for finalized objects.
We'd report for all objects or broke down per-collection/snapshot.

Second, we could report expensive finalizers, which ones when over X ms.